wrappers/rust/icicle-fields/icicle-goldilocks/src/lib.rs

```diff
@@ -3,7 +3,7 @@ pub mod ntt;
 pub mod polynomials;
 // TODO: uncomment once implemented for goldilocks
 // pub mod poseidon;
-// pub mod poseidon2;
+pub mod poseidon2;
```

Comment:
@LeonHibnik This should not be enabled until the following is added:
- Golidlocks supports poseidon2: https://github.com/ingonyama-zk/icicle/blob/main/icicle/cmake/features.cmake#L9
- Poseidon2 CPU backend supports using Goldilocks constants: https://github.com/ingonyama-zk/icicle/blob/main/icicle/backend/cpu/src/hash/cpu_poseidon2.cpp#L4-L33
- (optional) Poseidon2 CUDA backend supports using Goldilocks constants.